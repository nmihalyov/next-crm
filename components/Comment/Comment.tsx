import { Comment as CommentComponent, Avatar, Typography } from 'antd';

import type { Comment as CommentType } from '../../types/comment';

const Comment: React.FC<CommentType> = props => {
  const { name, body } = props;

  return (
    <CommentComponent
      author={<a>{name}</a>}
      avatar={<Avatar>{name.substr(0, 1).toUpperCase()}</Avatar>}
      content={<Typography.Paragraph>{body}</Typography.Paragraph>} />
  );
};

export default Comment;