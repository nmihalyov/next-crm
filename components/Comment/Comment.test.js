import Comment from './Comment';

describe('Comment:', () => {
  const setUp = props => shallow(<Comment {...props} />);
  const name = 'John Doe';
  const body = 'Comment body text'
  let component;

  beforeEach(() => {
    component = setUp({name, body});
  });

  it('renders CommentComponent', () => {
    const comment = component.find('Comment');

    expect(comment).toHaveLength(1);
  });

  it('renders author name\'s first letter', () => {
    const avatar = component.dive().find('Avatar');

    expect(avatar.text()).toBe(name.substr(0, 1).toUpperCase());
  });

  it('renders body text', () => {
    const paragraph = component.dive().find('Paragraph');

    expect(paragraph.children().text()).toBe(body);
  });
});