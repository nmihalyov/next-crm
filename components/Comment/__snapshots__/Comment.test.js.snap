// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment: renders Comment 1`] = `
<Comment
  author={
    <a>
      John Doe
    </a>
  }
  avatar={
    <Avatar
      shape="circle"
      size="default"
    >
      J
    </Avatar>
  }
  content={
    <Paragraph>
      Comment body text
    </Paragraph>
  }
/>
`;
